tokio-fs/src/blocking_pool.rs

```diff
@@ -0,0 +1,83 @@
+use futures::future;
+use futures::prelude::*;
+use futures::sync::mpsc;
+use futures::sync::oneshot;
+
+use std::any::Any;
+use std::panic;
+use std::thread;
+
+lazy_static! {
+    /// A thread pool for executing blocking functions.
+    static ref BLOCKING_POOL: tokio_threadpool::ThreadPool = {
+        tokio_threadpool::Builder::new().pool_size(1).build()
+    };
+
+    /// Schedules blocking functions for execution on the thread pool.
+    static ref SENDER: mpsc::UnboundedSender<Box<FnMut() + Send + 'static>> = {
+        let (tx, rx) = mpsc::unbounded();
+
+        BLOCKING_POOL.spawn(rx.for_each(|mut f: Box<FnMut() + Send + 'static>| {
+            BLOCKING_POOL.spawn(future::poll_fn(move || {
+                tokio_threadpool::blocking(|| f()).map_err(|_| ())
```

Comment:
It looks like, for each blocking op, a `Box<FnMut() ...>` is allocated and a task is allocated (when spawning for the op). 

Do you have thoughts on how to reduce this?